sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubUnboundedSink.java

```diff
@@ -220,6 +234,16 @@ public void populateDisplayData(DisplayData.Builder builder) {
   /** Publish messages to Pubsub in batches. */
   @VisibleForTesting
   static class WriterFn extends DoFn<Iterable<OutgoingMessage>, Void> {
+    private class OutgoingData {
+      int bytes;
+      List<OutgoingMessage> messages;
+
+      OutgoingData() {
+        this.bytes = 0;
+        this.messages = new ArrayList<>(publishBatchSize);
```

Comment:
I don't think we want to reserve full batch size. if we have unique ordering keys we're adding 1K overhead to each one.  Seems ok just to use normal array list growth to handle appending